api/src/main/java/org/pmiops/workbench/api/CohortAnnotationDefinitionController.java

```diff
@@ -74,7 +76,13 @@ public CohortAnnotationDefinition apply(org.pmiops.workbench.db.model.CohortAnno
         request.setCohortId(cohortId);
 
         org.pmiops.workbench.db.model.CohortAnnotationDefinition cohortAnnotationDefinition =
-        cohortAnnotationDefinitionDao.save(FROM_CLIENT_COHORT_ANNOTATION_DEFINITION.apply(request));
+                FROM_CLIENT_COHORT_ANNOTATION_DEFINITION.apply(request);
+
+        try {
+            cohortAnnotationDefinition = cohortAnnotationDefinitionDao.save(cohortAnnotationDefinition);
+        } catch (DataIntegrityViolationException e) {
+            throw new ConflictException(e.getMessage());
```

Comment:
A couple things:

- Many things can cause DataIntegrityViolationException.
- This exception doesn't really tell the client what went wrong.

That said, if we think the client can be responsible for enforcing uniqueness of annotation definition names, and we just want this as a backstop, this seems OK.

The alternative is to do name uniqueness checking in code explicitly, and throw an exception with an ErrorResponse like:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/exceptions/BadRequestException.java#L16

and have the client parse it to figure out what went wrong.

@johnstill

Reply:
throwing a new `ConflictException` with the original's message is almost certainly right if the integrity violation arose from a uniqueness constraint violation - I guess the question is what other scenarios would result in an integrity violation here and should we worry about them?

Throwing a BadRequest is definitely wrong.  `BadRequest` to me means the message body could not be parsed or I the client made some other error in the request itself, not that my request is syntactically valid but violates a database constraint.

Reply:
I can check the name uniqueness explicitly, then throw a new ConflictException with a better message about unique constraint. Sound ok @danqrodney @johnstill ?

Reply:
Sounds good to me.

Reply:
SGTM. I'm also OK with this code; I just want to make sure the client is going to do something intelligent and say "you already used that name". That could be done by having the client check for uniqueness, or it could be returning a response that makes it clear that a uniqueness constraint was violated (probably by returning an ErrorResponse).